File: 191-number-of-1-bits/191-number-of-1-bits.java
public class Solution {
    // you need to treat n as an unsigned value
    
    // Time = O(1)
    // Space = O(1)
    public int hammingWeight(int n) {
        int num_ones = 0;
        int mask = 1;
            
        for (int i = 0; i < 32; i++) {
            if ((n & mask) != 0) {
                num_ones++;
            }
            mask = mask << 1;
        }
        
        return num_ones;
    }
}

File: 191-number-of-1-bits/191-number-of-1-bits.py
class Solution:
    # Time = O(1) -> fixed length of bits (i.e. at most 32-bits given for a number)
    # Space = O(1)
    def hammingWeight(self, n: int) -> int:
        num_ones = 0
        num = n
        bitmask = 1
        while num > 0:
            if num & bitmask == 1:
                num_ones += 1
            num = num >> 1
        return num_ones
        